en/guides/microservice-mongodb.md

```diff
@@ -0,0 +1,547 @@
+                    !migratedDatabaseSchemas.Contains(connectionString)) //the database was not migrated yet
+                {
+                    _logger.LogInformation($"Migrating tenant database: {tenant.Name} ({tenant.Id})");
+                    await MigrateAllDatabasesAsync(tenant.Id, cancellationToken);
+                    migratedDatabaseSchemas.AddIfNotContains(connectionString);
+                }
+
+                //Seed data
+                _logger.LogInformation($"Seeding tenant data: {tenant.Name} ({tenant.Id})");
+                await SeedDataAsync();
+            }
+        }
+    }
+
+    private async Task MigrateAllDatabasesAsync(
+        Guid? tenantId,
+        CancellationToken cancellationToken)
+    {
+        using (var uow = _unitOfWorkManager.Begin(requiresNew: true, isTransactional: false))
+        {
+            var dbContexts = _serviceProvider.GetServices<IAbpMongoDbContext>();
+
+            var connectionStringResolver = _serviceProvider.GetRequiredService<IConnectionStringResolver>();
+
+            if (tenantId != null)
+            {
+                dbContexts = dbContexts.Where(x => !x.GetType().IsDefined(typeof(IgnoreMultiTenancyAttribute)));
+            }
+
+            foreach (var dbContext in dbContexts)
+            {
+                var connectionString =
+                    await connectionStringResolver.ResolveAsync(
+                        ConnectionStringNameAttribute.GetConnStringName(dbContext.GetType()));
+                if (connectionString.IsNullOrWhiteSpace())
+                {
+                    continue;
+                }
+                var mongoUrl = new MongoUrl(connectionString);
+                var databaseName = mongoUrl.DatabaseName;
+                var client = new MongoClient(mongoUrl);
+
+                if (databaseName.IsNullOrWhiteSpace())
+                {
+                    databaseName = ConnectionStringNameAttribute.GetConnStringName(dbContext.GetType());
+                }
+
+                (dbContext as AbpMongoDbContext)?.InitializeCollections(client.GetDatabase(databaseName));
+            }
+
+            await uow.CompleteAsync(cancellationToken);
+        }
+
+        _logger.LogInformation(
+            $"All databases have been successfully migrated ({(tenantId.HasValue ? $"tenantId: {tenantId}" : "HOST")}).");
+    }
+
+    private async Task SeedDataAsync()
+    {
+        await _dataSeeder.SeedAsync(
+            new DataSeedContext(_currentTenant.Id)
+                .WithProperty(IdentityDataSeedContributor.AdminEmailPropertyName,
+                    IdentityServiceDbProperties.DefaultAdminEmailAddress)
+                .WithProperty(IdentityDataSeedContributor.AdminPasswordPropertyName,
+                    IdentityServiceDbProperties.DefaultAdminPassword)
+        );
+    }
+}
+```
+
+### DatabaseMigrationEventHandlerBase
+
+Update `DatabaseMigrationEventHandlerBase` class to the following:
```

Comment:
I am not sure about changing the base handler to mongodb... This will cause other services compilation error. Need to find a better solution.

Reply:
We need to update the DatabaseMigrationEventHandler of all services.
https://github.com/abpio/abp-commercial-docs/pull/146/files#diff-f196f1f113682c06907674d432efffec29249647b362f3aa1e79f717bc8012d5R488